Liya-w2-react

liya-oz opened the pull request.

sycons reviewed (changes requested):
Hi Liya, there are 2 versions for `src` folder and `package.json` file in this PR. One is in the `project` folder and the other is in the `ecommerce` folder. Basically there are 2 React Apps in this PR. Can you clean this up so that there is only 1 React App and I can continue the review on that app?

Can you also add the link to the deployed app (somewher like netlify) to this PR? 

Thank you.

liya-oz commented:
Hi @sycons!

I made the changes you requested:
- Removed one of the React apps and kept just necessary one
- Added the link to the deployed app on [Netlify](https://vocal-madeleine-0aa54d.netlify.app)

Please have a look! Thank you!

sycons commented on week2/project/ecommerce/src/App.css:
```diff
@@ -0,0 +1,219 @@
+  transition: transform 0.2s;
+}
+
+.product-image-container {
```
**Optional change:** 
The `App.css` file can get quite big over time. It order to keep it manageable, it's best to keep styles that will be used in more than one component here and styles that are specific to only one component can be placed in a `.css` file at the same level as the component and access by importing.
E.g. The styling for `product-image-container` is only used in `ProductItem.jsx`. This can be moved to a file names `ProductItem.css` and then accessed in the component by adding the line `import `./ProductItem.css'`

sycons commented on week2/project/ecommerce/src/App.jsx:
```diff
@@ -0,0 +1,48 @@
+
+  return (
+    <div>
+      {location.pathname === '/' && (
```
Instead of using `location.pathname === '/'` to ensure this is only displayed on the home page, a better way to do this is to move it inside the `Route` for `/` i.e. line 33

sycons replied:
@liya-oz Can you implement this change please? Only this change is required for approval.

sycons commented on week2/project/ecommerce/src/App.jsx:
```diff
@@ -0,0 +1,48 @@
+        <Route
+          path="/"
+          element={
+            <div className="app-header">
```
Line 18 to 23 can be moved here.

sycons commented on week2/project/ecommerce/src/utility.js:
```diff
@@ -0,0 +1,21 @@
+// Normalize category names by removing 'FAKE:' prefix and trimming extra spaces
+export function cleanCategory(category) {
```
**Optional change:**
Have a look if these utlities are used anywhere or needed anymore. If they are not, they can be removed. E.g. The API doesn't return categories or product titles with `FAKE:` in them.

sycons reviewed (changes requested):
Following was implemented:
- `fake-data` directory is no longer used ✅
- App connects to endpoints in the API to fetch data for categories and products ✅
- Filter the products in the API and not in the frontend ✅
- App displays loading spinner when waiting for request to complete fetching data ✅
- App shows an error message if the request failed ✅
- Clicking on product card goes to a detail page ✅
- Detail page fetches data from an endpoint ✅
- Deploy app somewhere (like netlify) and add link to PR ✅

Compliments:
The stlying of this app is sleek and smooth. It responds beautifully for different endpoints, with the padding and the layout holding really well ⭐ 

Some optional changes have been suggested. These are not a blocker for approval.

Only the change requested in `App.jsx` is required to approve this PR.

liya-oz commented:
Hi @sycons! 😊  
Thank you so much for your feedback!  I've implemented both optional changes and the updates in `App.jsx`.

sycons reviewed (approved):
Amazing work Liya! ⭐ 

I appreciate you implementing even the optional changes. The code is a lot more polished and well structured now.
